### PR TITLE
Add PropertyTypes to SSM Resource Specification patching

### LINKED ISSUE #1695: Exceptions in E3002 and E3003 when evaluating AWS::StepFunctions::StateMachine.TagsEntry for eu-west-2
Given the following template:

```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: AWS::Serverless-2016-10-31

Resources:
  StateMachineRole:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Version: '2012-10-17'
        Statement:
          - Effect: Allow
            Principal:
              Service: states.amazonaws.com
            Action: sts:AssumeRole
  
  StateMachine:
    Type: AWS::Serverless::StateMachine
    Properties:
      Definition:
        StartAt: Example
      Role: !GetAtt StateMachineRole.Arn
```

The following linting errors are output when using cfn-lint 0.35.1 with the eu-west-2 region specified:

```
E0002 Unknown exception while processing rule E3002: 'AWS::StepFunctions::StateMachine.TagsEntry'
example.yml:1:1

E0002 Unknown exception while processing rule E3003: 'AWS::StepFunctions::StateMachine.TagsEntry'
example.yml:1:1
```

The errors are not shown when specifying the region as eu-west-1 or us-east-1.

### PR DESCRIPTION
**Pull request by kddejong**
*Issue #, if available:*
fix #1695 
*Description of changes:*
- Add StateMachine property types to spec patching to eu-west-2

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by PatMyron — DISMISSED** [review]
Won't this file just get overwritten? Think updating [`update_specs_services_from_ssm.py`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/scripts/update_specs_services_from_ssm.py) is the better way forward

**Comment by kddejong** [comment]
Ugh... you're right.  My fault. Let me fix this

**Comment by PatMyron** [comment]
ratio of `PropertyTypes` to `ResourceTypes` seems to roughly line up across the board now:
```bash
curl -s --compressed https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json | jq '.ResourceTypes | length'
565
curl -s --compressed https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json | jq '.PropertyTypes | length'
1549

grep 'ResourceTypes' src/cfnlint/data/ExtendedSpecs/*-*/06_ssm_service_removal.json | wc -l    
     172
grep 'PropertyTypes' src/cfnlint/data/ExtendedSpecs/*-*/06_ssm_service_removal.json | wc -l
     527

grep 'ResourceTypes' src/cfnlint/data/ExtendedSpecs/*-*/07_ssm_service_addition.json | wc -l
     755
grep 'PropertyTypes' src/cfnlint/data/ExtendedSpecs/*-*/07_ssm_service_addition.json | wc -l
    1998
```

File: scripts/update_specs_services_from_ssm.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 service_map = {
     'acm': ['AWS::CertificateManager::'],
     'apigateway': ['AWS::ApiGateway::', 'AWS::ApiGatewayV2::'],
-    'application-autoscaling': ['AWS::ApplicationAutoScaling::'],
+    # 'application-autoscaling': ['AWS::ApplicationAutoScaling::'], ## remove because SSM endpoints aren't correct
```
Comment (PatMyron) [inline scripts/update_specs_services_from_ssm.py:27]: how do we handle these long term? does this happen often? is there someone we can reach out to about these? are they usually fixed quickly?

Comment (kddejong) [reply scripts/update_specs_services_from_ssm.py:27]: This is the first time I have noticed an issue with this data and I'm not sure where to take it yet.

Comment (kddejong) [reply scripts/update_specs_services_from_ssm.py:27]: Submitted a support ticket.  I'll update the script once the issue is resolved.

**Pull request merged**